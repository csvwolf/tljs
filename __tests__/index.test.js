import tl from '../src';
import fixturesTest from '../fixtures';

describe('tl', () => {
  it('should return a string', () => {
    const t = tl`hello`;
    expect(t()).toBe('hello');
  });

  it('should escape new line', () => {
    const t = tl`hello\nworld`;
    expect(t()).toBe('hello\nworld');
  });

  it('should escape double quote', () => {
    const t = tl`"hello" \"world\"`;
    expect(t()).toBe('"hello" "world"');
  });

  it('should support embedded javascript', () => {
    const t = tl`
      <div>
      ${' if (title) { '}
        <h1>${' echo(title) '}</h1>
      ${' } '}
        <hr/>
      ${' posts.forEach(post => { '}
        <p>${' echo(post.content) '}</p>
      ${' }) '}
      </div>
    `;
    expect(t({
      title: 'title',
      posts: [
        { content: 'are' },
        { content: 'you' },
        { content: 'ok' }
      ]
    })).toMatchSnapshot();
  });

  it('should include another template', () => {
    expect(fixturesTest({
      valA: 'value_a',
      valB: 'value_b',
    })).toMatchSnapshot();
  });
});
