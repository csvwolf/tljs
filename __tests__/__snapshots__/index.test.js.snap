// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tl should include another template 1`] = `
"
  <div>
    <h1>A</h2>
    <p>value_a</p>
    
  <p>
    <h3>B</h3>
    <span>value_b</span>
  </p>

  </div>
"
`;

exports[`tl should support embedded javascript 1`] = `
"
      <div>
      
        <h1>title</h1>
      
        <hr/>
      
        <p>are</p>
      
        <p>you</p>
      
        <p>ok</p>
      
      </div>
    "
`;
